pkg/manager/options.go

```diff
@@ -138,8 +139,8 @@ func (o *Options) defaults() {
 
 	if o.Username == "" || o.Password == "" {
 		credentials := o.getCredentials()
-		o.Username = credentials["username"]
-		o.Password = credentials["password"]
+		o.Username = escapeCredentials(credentials["username"])
```

Comment:
actually, while playing with this PR I remembered that single quoting here https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/config/default/credentials.yaml#L9 would do the trick. so that might be simpler than what we were thinking

Reply:
Yes, this approach seems a bit brittle. We have funny regex escaping for the ini files because INI files are not defined in a standard and the Go library that was being used had a bespoke interpretation. Otherwise we should rely on the YAML specification or the appropriate library escaping, otherwise we'll be forever chasing edge cases.

Reply:
fun thing: printing/logging actually escapes before printing, so it can be difficult to know what's ends up in memory

Reply:
>fun thing: printing/logging actually escapes before printing, so it can be difficult to know what's ends up in memory

@yastij This is what I was talking about on Slack yesterday. I will update the PR with the single quotes and modify the test.

Reply:
ack, thank you @srm09 !